proctor-common/src/main/java/com/indeed/proctor/common/ForceGroupsOptionsStrings.java

```diff
@@ -38,39 +47,156 @@ public static ForceGroupsOptions parseForceGroupsString(@Nullable final String f
         final String[] pieces = forceGroupsString.split(",");
         // detect integer number from end of string
         for (final String rawPiece : pieces) {
-            final String piece = rawPiece.trim();
-            if (piece.isEmpty()) {
+            // split string to separate force group and payload
+            final String[] bucketAndPayloadValuesStr = rawPiece.split(";", FORCE_PARAMETER_MAX_SIZE);
+            final String groupString = bucketAndPayloadValuesStr[FORCE_PARAMETER_BUCKET_IDX].trim();
+            if (groupString.isEmpty()) {
                 continue;
             }
 
-            ForceGroupsDefaultMode.fromToken(piece)
+            ForceGroupsDefaultMode.fromToken(groupString)
                     .ifPresent(builder::setDefaultMode);
 
-            int bucketValueStart = piece.length() - 1;
-            for (; bucketValueStart >= 0; bucketValueStart--) {
-                if (!Character.isDigit(piece.charAt(bucketValueStart))) {
-                    break;
+            final Optional<Integer> bucketValueStart = getBucketValueStart(groupString);
+
+            if (bucketValueStart.isPresent()) {
+                //  bucketValueStart should now be the index of the minus sign or the first digit in a run of digits going to the end of the word
+                final String testName = groupString.substring(0, bucketValueStart.get()).trim();
+                final String bucketValueStr = groupString.substring(bucketValueStart.get()).trim();
+                try {
+                    final Integer bucketValue = Integer.valueOf(bucketValueStr);
+                    builder.putForceGroup(testName, bucketValue);
+                    if (bucketAndPayloadValuesStr.length == FORCE_PARAMETER_MAX_SIZE) {
+                        final Payload payloadValue = parseForcePayloadString(bucketAndPayloadValuesStr[FORCE_PARAMETER_PAYLOAD_IDX]);
+                        if (payloadValue != null) {
+                            builder.putForcePayload(testName, payloadValue);
+                        }
+                    }
+                } catch (final NumberFormatException e) {
+                    LOGGER.error("Unable to parse bucket value " + bucketValueStr + " as integer", e);
                 }
             }
-            //  if no name or no value was found, it's not a valid proctor test bucket name
-            if ((bucketValueStart == piece.length() - 1) || (bucketValueStart < 1)) {
-                continue;
-            }
-            //  minus sign can only be at the beginning of a run
-            if (piece.charAt(bucketValueStart) != '-') {
-                bucketValueStart++;
+        }
+        return builder.build();
+    }
+
+    private static Optional<Integer> getBucketValueStart(final String groupString) {
+        int bucketValueStart = groupString.length() - 1;
```

Comment:
i know this is from the original code. but is there a formal limit such as bucket value can only be a single digit?